crates/bevy_tasks/src/bevy_executor.rs

```diff
@@ -0,0 +1,1296 @@
+                                    return Poll::Pending;
+                                }
+                            }
+                            Some(r) => {
+                                // Wake up.
+                                self.wake();
+
+                                // Notify another ticker now to pick up where this ticker left off, just in
+                                // case running the task takes a long time.
+                                self.state.notify();
+
+                                return Poll::Ready(r);
+                            }
+                        }
+                    }
+                }).unwrap_or(Poll::Pending)
+            }
+        })
+    }
+}
+
+impl Drop for Ticker<'_> {
+    fn drop(&mut self) {
+        // If this ticker is in sleeping state, it must be removed from the sleepers list.
+        if self.sleeping != 0 {
+            let mut sleepers = self.state.sleepers.lock().unwrap_or_else(PoisonError::into_inner);
+            let notified = sleepers.remove(self.sleeping);
+
+            self.state
+                .notified
+                .store(sleepers.is_notified(), Ordering::Release);
+
+            // If this ticker was notified, then notify another ticker.
+            if notified {
+                drop(sleepers);
+                self.state.notify();
+            }
+        }
+    }
+}
+
+/// A worker in a work-stealing executor.
+///
+/// This is just a ticker that also has an associated local queue for improved cache locality.
+struct Runner<'a> {
+    /// The executor state.
+    state: &'a State,
+
+    /// Inner ticker.
+    ticker: Ticker<'a>,
+
+    /// Bumped every time a runnable task is found.
+    ticks: usize,
+
+    // The thread local state of the executor for the current thread.
+    local_state: &'a ThreadLocalState,
+}
+
+impl Runner<'_> {
+    /// Creates a runner and registers it in the executor state.
+    fn new(state: &State) -> Runner<'_> {
+        let local_state = THREAD_LOCAL_STATE.get_or_default();
+        let runner = Runner {
+            state,
+            ticker: Ticker::new(state),
+            ticks: 0,
+            local_state,
+        };
+        state
+            .stealer_queues
+            .write()
+            .unwrap_or_else(PoisonError::into_inner)
+            .push(&local_state.stealable_queue);
+        runner
+    }
+
+    /// Waits for the next runnable task to run.
+    fn runnable(&mut self, _rng: &mut fastrand::Rng) -> impl Future<Output = Runnable> {
+        // SAFETY: The provided search function does not access LOCAL_QUEUE in any way, and thus cannot 
+        // alias.
+        let runnable = unsafe {
+            self
+            .ticker
+            .runnable_with(|tls| {
+                if let Some(r) = tls.local_queue.pop_back() {
+                    return Some(r);
+                }
+
+                crate::cfg::multi_threaded! {
+                    if {
```

Comment:
Since the else clause is empty, this if shouldn't be needed. Pretty sure you can just do `crate::cfg::multi_threaded! { ... }`.